### ISSUE TITLE
random_split should support stratification column

### ISSUE THREAD

**Issue by CESARDELATORRE** [Azure/MachineLearningNotebooks#723]
random_split should support stratification column so you can split any dataset and make sure its split datasets are balanced based on a particular column. This is for instance important for datasets to be used by Binary Classification algorithms, so you make sure that training and test datasets are not imbalanced.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 5d7a9417-1b43-e707-b546-8eaa37a31e6d
* Version Independent ID: 1d9554bd-9be0-a026-f6ce-5477d51e3e9e
* Content: [azureml.data.TabularDataset class - Azure Machine Learning Python](https://docs.microsoft.com/en-us/python/api/azureml-core/azureml.data.tabulardataset?view=azure-ml-py#random-split-percentage--seed-none-)
* Content Source: [AzureML/docs-ref-autogen/azureml-core/azureml.data.TabularDataset.yml](https://github.com/MicrosoftDocs/MachineLearning-Python-pr/blob/live/AzureML/docs-ref-autogen/azureml-core/azureml.data.TabularDataset.yml)
* Service: **machine-learning**
* Sub-service: **core**
* GitHub Login: @j-martens
* Microsoft Alias: **jmartens**

**Comment by CESARDELATORRE** [comment]
As a comparison, stratification column is, by the way, supported by train_test_split() in Scikit-Learn.

**Comment by YutongTie-MSFT** [comment]
@CESARDELATORRE  Thanks for taking the time to provide feedback! This is an interesting feature idea and we want to make sure it gets directly into the hands of the product team. Fortunately, we've got a dedicated channel for that very purpose :) You can find it over here: https://feedback.azure.com/forums/257792

Thanks a lot! We will now proceed to close this thread. If there are further questions regarding this matter, please respond here and @YutongTie-MSFT and we will gladly continue the discussion.
